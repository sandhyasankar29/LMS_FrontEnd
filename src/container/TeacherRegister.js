import React, { useState } from 'react';
import axios from 'axios';
import {
    Box,
    Button,
    Form,
    FormField,
    Grommet,
    MaskedInput,
    TextInput,
    Grid,
    Select,
} from 'grommet';
import Swal from 'sweetalert2'
import { Hide, View } from 'grommet-icons';
import { grommet } from 'grommet/themes';
import { urlRegisterTeacher } from './Url';
export const TeacherRegister = () => {
    const daysInMonth = month => new Date(2021, month, 0).getDate();
    const [name, setName] = useState('');
    const [lastName, setlastName] = useState('');
    const [phNo, setPhoneNumber] = useState();
    const [email, setEmail] = useState('');
    const [rollNo, setRollNumber] = useState();
    const [doj, setDoj] = useState('');
    const [branch, setBranch] = useState();
    const [password, setPassword] = useState();
    const [reveal, setReveal] = useState("")
    const onSubmitSignUp = () => {
        if (name && lastName && email && branch && password && doj && phNo) {
            axios.post(urlRegisterTeacher, {
                tuserid: rollNo,
                lastname: lastName,
                firstname: name,
                doj: doj,
                branch: branch,
                email: email,
                password: password,
                phoneNumber: phNo
            }).then(res => {
                Swal.fire({
                    icon: 'success',
                    title: 'Successfully Registered',
                    text: 'Please Wait for Admin\'s Approval'
                })
            }).catch(error => {
                Swal.fire({
                    icon: 'error',
                    title: 'Registration Failed',
                    text: 'Please Check the Data'
                })
            })
        }
    }

    document.body.style.overflow = "hidden"
    return (
        <Grommet full theme={grommet}>
            <Grid
                rows={['xxsmall', 'medium']}
                columns={['medium', 'medium']}
                gap="small"
                areas={[
                    { name: 'left', start: [0, 1], end: [0, 1] },
                    { name: 'right', start: [1, 1], end: [1, 1] },
                ]}
            >

                <Box fill align="center" justify="center" gridArea="left" style={{ marginLeft: "50%" }}>
                    <Box width="medium">
                        <Form
                            onSubmit={onSubmitSignUp}
                        >
                            <FormField label="First Name" name="name">
                                <TextInput
                                    name="name"
                                    value={name}
                                    onChange={event => setName(event.target.value)}
                                    placeholder='Registered First Name'
                                />
                            </FormField>
                            <FormField label="Last Name" name="lastName">
                                <TextInput
                                    name="lastName"
                                    value={lastName}
                                    onChange={event => setlastName(event.target.value)}
                                    placeholder='Registered Last Name'
                                />
                            </FormField>
                            <FormField label="Email" name="email">
                                <MaskedInput
                                    name="email"
                                    placeholder='example@gmail.com'
                                    value={email}
                                    onChange={event => setEmail(event.target.value)}
                                />
                            </FormField>
                            <FormField label="Teacher ID" name="rollNo">
                                <MaskedInput
                                    name="rollNo"
                                    placeholder='xxDeptxxx'
                                    value={rollNo}
                                    onChange={event => setRollNumber(event.target.value)}
                                />
                            </FormField>
                            <Box direction="row" justify="between" margin={{ top: 'large' }}>
                                <Button data-testid="button" id="registerBtn" type="submit" label="Register" primary />
                            </Box>
                        </Form>
                    </Box>
                </Box>
                <Box fill align="center" justify="center" gridArea="right" style={{ marginLeft: "75%" }}>
                    <Box width="medium">
                        <Form
                            onSubmit={onSubmitSignUp}
                        >
                        </Form>
                        <FormField label="Phone Number" name="phNo" >
                            <MaskedInput
                                name="phNo"
                                mask={[
                                    {
                                        length: 10,
                                        regexp: /^[0-9]{1,10}$/,
                                        placeholder: 'Phone Number',
                                    },
                                ]}
                                value={phNo}
                                onChange={event => setPhoneNumber(event.target.value)}
                            />
                        </FormField>
                        <FormField label="Date of Joining" >
                            <MaskedInput
                                name="doj"
                                mask={[
                                    {
                                        length: 4,
                                        options: Array.from({ length: 28 }, (v, k) => 2021 - k),
                                        regexp: /^[1-2]$|^19$|^20$|^19[0-9]$|^20[0-9]$|^19[0-9][0-9]$|^20[0-9][0-9]$/,
                                        placeholder: 'yyyy',
                                    },
                                    { fixed: '-' },
                                    {
                                        length: [1, 2],
                                        options: Array.from({ length: 12 }, (v, k) => k + 1),
                                        regexp: /^1[0,1-2]$|^0?[1-9]$|^0$/,
                                        placeholder: 'mm',
                                    },
                                    { fixed: '-' },
                                    {
                                        length: [1, 2],
                                        options: Array.from(
                                            {
                                                length: daysInMonth(parseInt(doj.split('/')[0], 10)),
                                            },
                                            (v, k) => k + 1,
                                        ),
                                        regexp: /^[1-2][0-9]$|^3[0-1]$|^0?[1-9]$|^0$/,
                                        placeholder: 'dd',
                                    },

                                ]}
                                value={doj}
                                onChange={event => setDoj(event.target.value)}
                            />
                        </FormField>
                        <Box margin={{ top: 'medium' }} gap="large">
                            <FormField label="Branch" name="branch" required>
                                <Select
                                    id="branch"
                                    options={['CSE', 'ECE', 'EEE', 'EIE', 'MEE', 'AEE', 'CHE', 'PHY', "Choose"]}
                                    value={branch}
                                    defaultValue={"Choose"}
                                    disabled={["Choose"]}
                                    onChange={({ option }) => setBranch(option)}
                                />
                            </FormField>
                        </Box>
                        <FormField label="Preferred Password" >
                            <Box direction="row" justify="between" margin={{ top: 'xxsmall' }}>
                                <TextInput
                                    name="passsword"
                                    plain
                                    type={reveal ? 'text' : 'password'}
                                    value={password}
                                    onChange={event => setPassword(event.target.value)}
                                />
                                <Button
                                    icon={reveal ? <View size="medium" /> : <Hide size="medium" />}
                                    onClick={() => setReveal(!reveal)}
                                />
                            </Box>
                        </FormField>
                    </Box>
                </Box>
            </Grid>
        </Grommet>
    );
};



export default TeacherRegister;