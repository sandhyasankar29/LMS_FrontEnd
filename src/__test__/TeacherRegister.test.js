import React from 'react';
import ReactDOM from 'react-dom';
import TeacherRegister from '../container/TeacherRegister';
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TeacherRegister></TeacherRegister>, div)
})

it("matches snapshot", () => {
    const tree = renderer.create(<TeacherRegister></TeacherRegister>).toJSON();
    expect(tree).toMatchSnapshot();
})
