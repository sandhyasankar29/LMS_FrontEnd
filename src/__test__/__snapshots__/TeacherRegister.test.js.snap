// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 bWDPZI"
>
  <div
    className="StyledGrid-sc-1wofa1l-0 inzmHE"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 jKMlzI"
      style={
        Object {
          "marginLeft": "50%",
        }
      }
    >
      <div
        className="StyledBox-sc-13pk1d4-0 gfocoz"
      >
        <form
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="StyledText-sc-1sadyjn-0 drTEXR"
            >
              First Name
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 iuIygK"
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Registered First Name"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="StyledText-sc-1sadyjn-0 drTEXR"
            >
              Last Name
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 iuIygK"
                  name="lastName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Registered Last Name"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="StyledText-sc-1sadyjn-0 drTEXR"
            >
              Email
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
            >
              <div
                className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 dOerlP"
              >
                <input
                  autoComplete="off"
                  className="StyledMaskedInput-sc-99vkfa-0 gMZsTd"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="example@gmail.com"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="StyledText-sc-1sadyjn-0 drTEXR"
            >
              Teacher ID
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
            >
              <div
                className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 dOerlP"
              >
                <input
                  autoComplete="off"
                  className="StyledMaskedInput-sc-99vkfa-0 gMZsTd"
                  name="rollNo"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="xxDeptxxx"
                />
              </div>
            </div>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 LAyOB"
          >
            <button
              className="StyledButton-sc-323bzc-0 jfCvHs"
              data-testid="button"
              id="registerBtn"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="submit"
            >
              Register
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 cQVhiT"
      style={
        Object {
          "marginLeft": "75%",
        }
      }
    >
      <div
        className="StyledBox-sc-13pk1d4-0 gfocoz"
      >
        <form
          onReset={[Function]}
          onSubmit={[Function]}
        />
        <div
          className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Phone Number
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 dOerlP"
            >
              <input
                autoComplete="off"
                className="StyledMaskedInput-sc-99vkfa-0 gMZsTd"
                name="phNo"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Phone Number"
              />
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Date of Joining
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 dOerlP"
            >
              <input
                autoComplete="off"
                className="StyledMaskedInput-sc-99vkfa-0 gMZsTd"
                name="doj"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="yyyy-mm-dd"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 hNtvAO"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="StyledText-sc-1sadyjn-0 drTEXR"
            >
              Branch
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
            >
              <button
                aria-label="Open Drop"
                className="StyledButton-sc-323bzc-0 hkpJCj Select__StyledSelectDropButton-sc-17idtfo-1 dYfCep"
                id="branch"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                type="button"
              >
                <div
                  className="StyledBox-sc-13pk1d4-0 jmuHez"
                >
                  <div
                    className="StyledBox-sc-13pk1d4-0 iAryDB"
                  >
                    <div
                      className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
                    >
                      <input
                        autoComplete="off"
                        className="StyledTextInput-sc-1x30a0s-0 iuIygK Select__SelectTextInput-sc-17idtfo-0 bIurki"
                        id="branch__input"
                        onBlur={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        readOnly={true}
                        tabIndex="-1"
                        type="text"
                        value="Choose"
                      />
                    </div>
                  </div>
                  <div
                    className="StyledBox-sc-13pk1d4-0 eA-DBEo"
                    style={
                      Object {
                        "minWidth": "auto",
                      }
                    }
                  >
                    <svg
                      aria-label="FormDown"
                      className="StyledIcon-ofa7kd-0 fuviCN"
                      viewBox="0 0 24 24"
                    >
                      <polyline
                        fill="none"
                        points="18 9 12 15 6 9"
                        stroke="#000"
                        strokeWidth="2"
                      />
                    </svg>
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Preferred Password
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 hpZuiW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 hubMyb"
                  name="passsword"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  type="password"
                />
              </div>
              <button
                className="StyledButton-sc-323bzc-0 hiyJPE"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                type="button"
              >
                <svg
                  aria-label="Hide"
                  className="StyledIcon-ofa7kd-0 dEeaTZ"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,17 C9.27272727,17 6,14.2222222 6,12 C6,9.77777778 9.27272727,7 12,7 C14.7272727,7 18,9.77777778 18,12 C18,14.2222222 14.7272727,17 12,17 Z M11,12 C11,12.55225 11.44775,13 12,13 C12.55225,13 13,12.55225 13,12 C13,11.44775 12.55225,11 12,11 C11.44775,11 11,11.44775 11,12 Z M20,5 L4,19"
                    fill="none"
                    stroke="#000"
                    strokeWidth="2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
